styles.css

```diff
@@ -56,10 +56,26 @@ body {
 
 .result-selection {
     font-size: 2rem;
-    opacity: .5;
+    opacity: 0.6;
 }
 
 .result-selection.winner {
     font-size: 2.5rem;
     opacity: 1;
+}
+
+/* Proper Fade-out effect */
+
+/* as soon as there are 5 or more divs: */
+.history:has(.result-selection:nth-child(n+5)) {
+
+    /* styling the 5th and all following divs */
+    .result-selection:nth-of-type(n+5) {
+        opacity: 0.3;
+    }
+
+    /* styling the 9th and all following divs */
+    .result-selection:nth-of-type(n+9) {
+        opacity: 0.1;
+    }
```

Comment:
Love to see you playing around with pseudo selectors like `nth-of-type`! There are loads more, and honestly, you'd be best served with playing with each of them enough to understand what they're for and then promptly forgetting it except to know that they exist.

This is precisely the kind of thing that I don't use enough to have memorized, but then when I wanna use it, I'm aware of it enough to know what to google for, and boom, coding again.

And while there are ways to share variables across css and js, what you'll find instead is that when React is involved, and then possibly even with things like tailwind, you'll have a better mechanism for directly manipulating css based on things like index/UI state from the JS.

So if you're _really_ curious, you can go down that path of shared variables in something like sass or whatever is fancy these days, but truly, the sooner you feel confident enough to start with React, the sooner you really get to start your "job-ready" UI work 🤓

Reply:
Everything Matt said.

If I'm being honest, I needed to look these pseudo selectors up because I completely forgot about them and I'm not sure I've ever actually used them. I would say I've always been _ok_ at CSS but not amazing at it. Matt's better than I. I'm stoked that you're playing around with these and know that they exist though.

I'd make sure you know how grids work at a basic level and then probably say you are fully proficient in CSS and can probably turn your focus to learning Typescript / React (again echoing Matt's sentiment)

Reply:
> I think this could be addressed by sharing this variable (maxHistoryRows) from the .js file across the .css file.
I will need to learn how to do that.

There's a few ways to go about this, but ultimately the _right_ way to have variables in your js/css is usually to either use a CSS preprocessor like Sass or Less, or what I find much more common these days, to use a CSS in Javascript library like emotion or styled-components, or a full on React design system like Material UI or Chakra where you can pass styling directly as typesafe "props" to your React "Components".

I use a combo of emotion and Material UI at work right now, and haven't edited a raw CSS file for a long long time. All the CSS properties and stuff you're learning though is still relevant, it's just passed directly in through javascript through various means.